core/domain/src/main/java/com/google/samples/apps/nowinandroid/core/domain/GetSortedFollowableAuthorsStreamUseCase.kt

```diff
@@ -17,33 +17,40 @@
 package com.google.samples.apps.nowinandroid.core.domain
 
 import com.google.samples.apps.nowinandroid.core.data.repository.AuthorsRepository
+import com.google.samples.apps.nowinandroid.core.data.repository.NewsRepository
 import com.google.samples.apps.nowinandroid.core.domain.model.FollowableAuthor
 import javax.inject.Inject
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.combine
 
 /**
  * A use case which obtains a list of authors sorted alphabetically by name with their followed
  * state.
  */
 class GetSortedFollowableAuthorsStreamUseCase @Inject constructor(
+    private val newsRepository: NewsRepository,
     private val authorsRepository: AuthorsRepository
 ) {
     /**
      * Returns a list of authors with their associated followed state sorted alphabetically by name.
      *
-     * @param followedTopicIds - the set of topic ids which are currently being followed.
+     * @param followedAuthorIds - the set of topic ids which are currently being followed.
      */
     operator fun invoke(followedAuthorIds: Set<String>): Flow<List<FollowableAuthor>> {
-        return authorsRepository.getAuthorsStream().map { authors ->
+        return combine(
```

Comment:
We should probably do this in SQL in the `AuthorDao`. The `news_resources_authors` table has all the information we need to make this query efficient.

```
SELECT * FROM authors
WHERE id in (
    SELECT  author_id
    FROM news_resources_authors
    GROUP BY author_id
    ORDER BY COUNT(author_id) DESC;
)
```

Reply:
The onboarding can then have its own model and each author can have a little badge showing the amount of articles they've written.